src/VscPreviewExtension/src/extension.ts

```diff
@@ -0,0 +1,269 @@
+                previewResult += tmp;
+            }
+            // The result be cut and this is the last one
+            else {
+                previewResult += tmp;
+                isEnd = true;
+                provider.update(documentUri);
+            }
+        });
+
+        this._spawn.stderr.on('data', function (data) {
+            console.log("error " + data + '\n');
+        });
+
+        this._spawn.on('exit', function (code) {
+            console.log('child process exit with code ' + code);
+        });
+    }
+
+    private sendtext() {
+        let editor = window.activeTextEditor;
+        if (!editor) {
+            return;
+        }
+        let doc = editor.document;
+        let docContent = doc.getText();
+        let filename = doc.fileName;
+        let rtPath = workspace.rootPath;
+        let filePath;
+        if (!rtPath) {
+            let indexOfFilename = filename.lastIndexOf('\\');
+            rtPath = filename.substr(indexOfFilename - 1);
+            filePath = filename.substring(0, indexOfFilename);
+        }
+        else {
+            let rtpath_length = rtPath.length;
+            filePath = filename.substr(rtpath_length + 1, filename.length - rtpath_length);
+        }
+        if (doc.languageId === "markdown") {
+            let numOfRow = doc.lineCount;
+            this._spawn.stdin.write(rtPath + '\n');
+            this._spawn.stdin.write(filePath + '\n');
+            this._spawn.stdin.write(numOfRow + '\n');
+            this._spawn.stdin.write(docContent + '\n');
+        }
+    }
+
+    public callDfm() {
+        if (this._isFirstTime) {
+            // In the firt time, if wait for the timeout, activeTextEditor will be translate to the preview window.
+            this._isFirstTime = false;
+            this.sendtext();
+        }
+        else if (!this._waiting) {
+            this._waiting = true;
+            setTimeout(() => {
+                this._waiting = false;
+                this.sendtext();
+            }, 300);
+        }
+    }
+}
+
+class MDDocumentContentProvider implements TextDocumentContentProvider {
+    private _context: ExtensionContext;
+    private _onDidChange = new EventEmitter<Uri>();
+    private _waiting: boolean;
+
+    constructor(context: ExtensionContext) {
+        this._context = context;
+        this._waiting = false;
+    }
+
+    private getMediaPath(mediaFile): string {
+        return this._context.asAbsolutePath(path.join('media', mediaFile));
+    }
+
+    public provideTextDocumentContent(uri: Uri): Thenable<string> {
+        return workspace.openTextDocument(Uri.parse(uri.query)).then(document => {
+            const head = [].concat(
+                '<!DOCTYPE html>',
+                '<html>',
+                '<head>',
+                '<meta http-equiv="Content-type" content="text/html;charset=UTF-8">',
+                `<link rel="stylesheet" type="text/css" href="${this.getMediaPath('tomorrow.css')}" >`,
```

Comment:
refer to [https://github.com/chriskempson/tomorrow-theme](https://github.com/chriskempson/tomorrow-theme), close.

---
In reply to: [75239337](https://github.com/dotnet/docfx/pull/616#discussion_r75239337) [](ancestors = 75239337,75132804)